dell-bios-fan-control: init package at 0-unstable-2022-01-19 and init module

TotalyEnglizLitrate opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Adds a package for `dell-bios-fan-control`, a utility to disable BIOS control of fans on some Dell laptops, allowing userspace fan control tools like `fancontrol` to manage fan speeds.

Adds a NixOS module that enables two services created using https://aur.archlinux.org/packages/dell-bios-fan-control-git as reference:
- `dell-bios-fan-control.service`: Disables BIOS fan control at boot
- `dell-bios-fan-control-resume.service`: Re-applies the setting after system resume

## Things done
- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [x] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

TotalyEnglizLitrate committed:
maintainers: add totalyenglizlitrate

TotalyEnglizLitrate committed:
dell-bios-fan-control: init at 20220119

.

TotalyEnglizLitrate committed:
nixos/dell-bios-fan-control: init module

TotalyEnglizLitrate commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 465772`
Commit: `baae01e42c37c2f41040e04712026639a18b3bb2`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>dell-bios-fan-control</li>
  </ul>
</details>

sarahec commented on pkgs/by-name/de/dell-bios-fan-control/package.nix:
```diff
@@ -0,0 +1,32 @@
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dell-bios-fan-control";
+  version = "20220119"; # upstream does not use versioning - use date of last commit
```
The version should be `0-unstable-20220119` per https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#versioning

pyrox0 replied:
nit: if it was date of commit, per versioning it would be `0-unstable-2022-01-19`, but we're splitting hairs at this point.

sarahec commented on pkgs/by-name/de/dell-bios-fan-control/package.nix:
```diff
@@ -0,0 +1,32 @@
+{
+  lib,
+  stdenv,
+  fetchgit,
```
`fetchFromGitHub,`

sarahec reviewed (changes requested):
Nice work. There's a few things you need to update.

sarahec commented:
Welcome :) Nice PR!

"style(nixfmt): satisfy the lint checker" should be squashed into the appropriate commit instead. (To spare you this hassle next time, remember to run `nix fmt` followed by the file path.

tomodachi94 commented on pkgs/by-name/de/dell-bios-fan-control/package.nix:
```diff
@@ -0,0 +1,32 @@
+    hash = "sha256-3ihzvwL86c9VJDfGpbWpkOwZ7qU0E5U2UuOeCwPMR1s=";
+  };
+
+  buildPhase = ''make'';
```
This line isn't necessary and can be deleted, `mkDerivation` invokes `make` during `buildPhase` by default.

TotalyEnglizLitrate committed:
dell-bios-fancontrol: Make requested changes

- Use `fetchFromGitHub` instead of `fetchgit`
- Modify version to adhere to convention
- remove buildPhase as it calls make by default

sarahec reviewed (changes requested):
Please squash the fourth commit (make changes) into the init commit. You can use `git rebase -i` to do this.

Otherwise LGTM (looks good to me)

sarahec commented:
> I- should not have been working on this half asleep 😭
> 
> edit: welp looks like I messed it up beyond the point of fix-ability 🤦, should I just create a new PR?

Yes, go ahead. 

It's all learning.
